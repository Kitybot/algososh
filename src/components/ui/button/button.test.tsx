import React from 'react';
import renderer from 'react-test-renderer';
import { Button } from './button';
import { render, screen, fireEvent } from '@testing-library/react';

describe('Рендеринг Кнопки', () => {

  it('Кнопка с текстом', () => {
    const tree = renderer.create(<Button text='Кнопка' type='button'/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Кнопка без текста', () => {
    const tree = renderer.create(<Button type='button'/>).toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('Заблокированная кнопка', () => {
    const tree = renderer.create(<Button type='button' disabled={true}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Кнопка с активным лоадером', () => {
    const tree = renderer.create(<Button type='button' isLoader={true}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Вызов колбэка при нажатии на кнопку', () => {
    window.alert = jest.fn();
    render(
      <Button 
        type='button' 
        text='Кнопка' 
        onClick={() => {alert('Кнопка нажата')}}
      />
    );
    const button = screen.getByText('Кнопка');
    fireEvent.click(button);
    expect(window.alert).toHaveBeenCalledWith('Кнопка нажата');
  })
});