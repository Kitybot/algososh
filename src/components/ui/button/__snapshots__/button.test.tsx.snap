// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Рендеринг Кнопки Заблокированная кнопка 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Рендеринг Кнопки Кнопка без текста 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Рендеринг Кнопки Кнопка с активным лоадером 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Рендеринг Кнопки Кнопка с текстом 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Кнопка
  </p>
</button>
`;